m4a_hq_mixer.s

```diff
@@ -717,101 +711,41 @@ C_fast_mixing_creation_loop:
     ADD     LR, LR, #(ARM_OP_LEN*38)
     STMIA   LR, {R2, R8-R10}
     SUB     LR, LR, #(ARM_OP_LEN*32)
-    ADDS    R5, R5, #0x40000000         @ do that for 4 blocks
+    ADDS    R5, R5, #0x40000000         @ do that for 4 blocks (unused pointer bits)
     BCC     C_fast_mixing_creation_loop
 
 C_skip_fast_mixing_creation:
     LDR     R8, [SP]                        @ restore R8 with the frame length
     LDR     R8, [R8, #(ARG_FRAME_LENGTH + 0x8 + 0xC)]
-    MOV     R2, #0xFF000000                 @ load the fine position overflow bitmask
+    MOVS    R2, #0xFF000000                 @ load the fine position overflow bitmask, set NE
```

Comment:
ohhh, interesting, very nice solution to make the MUL below conditional.

Reply:
The loop repeats with `bgt`, and `gt` implies `ne`.